include/picongpu/plugins/output/images/Visualisation.hpp

```diff
@@ -157,286 +159,654 @@ struct typicalFields < 5 >
+                );
+
+                bool const isCellOnSlice = IsPartOfSlice< >{}(
+                    realCell,
+                    sliceDim,
+                    localDomainOffset,
+                    slice
+                );
+
+                /* if the virtual worker is not calculating a cell out of the
+                 * selected slice than exit
+                 */
+                if( !isCellOnSlice )
+                    return;
+
+                // set fields of this cell to vars
+                typename T_BBox::ValueType field_b = fieldB( cellOffset );
+                typename T_EBox::ValueType field_e = fieldE( cellOffset );
+                typename T_JBox::ValueType field_j = fieldJ( cellOffset );
+
+                // multiply with the area size of each plane
+                field_j *= float3_X::create( CELL_VOLUME ) / cellSize;
+
+                /* reset picture to black
+                 *   color range for each RGB channel: [0.0, 1.0]
+                 */
+                float3_X pic(
+                    // typical values of the fields to normalize them to [0,1]
+                    visPreview::preChannel1(
+                        field_b / typicalFields< EM_FIELD_SCALE_CHANNEL1 >::get( ).x( ),
+                        field_e / typicalFields< EM_FIELD_SCALE_CHANNEL1 >::get( ).y( ),
+                        field_j / typicalFields< EM_FIELD_SCALE_CHANNEL1 >::get( ).z( )
+                    ),
+                    visPreview::preChannel2(
+                        field_b / typicalFields< EM_FIELD_SCALE_CHANNEL2 >::get( ).x( ),
+                        field_e / typicalFields< EM_FIELD_SCALE_CHANNEL2 >::get( ).y( ),
+                        field_j / typicalFields< EM_FIELD_SCALE_CHANNEL2 >::get( ).z( )
+                    ),
+                    visPreview::preChannel3(
+                        field_b / typicalFields< EM_FIELD_SCALE_CHANNEL3 >::get( ).x( ),
+                        field_e / typicalFields< EM_FIELD_SCALE_CHANNEL3 >::get( ).y( ),
+                        field_j / typicalFields< EM_FIELD_SCALE_CHANNEL3 >::get( ).z( )
+                    )
+                );
+
+                // draw to (perhaps smaller) image cell
+                image( imageCell ) = pic;
+            }
+        );
     }
 };
 
+/** derives two dimensional field from a particle slice
+ *
+ * The shape of a particle is not taken in account.
+ *
+ * @tparam T_numWorkers number of workers
+ */
+template< uint32_t T_numWorkers >
 struct KernelPaintParticles3D
 {
+    /** derive particle values
+     *
+     * @tparam T_ParBox pmacc::ParticlesBox, particle box type
+     * @tparam T_Mapping mapper functor type
+     * @tparam T_Acc alpaka accelerator type
+     *
+     * @param acc alpaka accelerator
+     * @param pb particle memory
+     * @param image[in,out] two dimensional image (without guarding cells)
+     * @param transpose indices to transpose dimensions range per dimension [0,simDim)
```

Comment:
QUESTION:
as above regarding the brackets

Reply:
fixed